App/src/Repository/UserRepository.php

```diff
@@ -30,4 +30,49 @@ public function findOneByname(string $name): ?User
 
         return $user;
     }
+
+    public function isUsernameAlreadyInUsed(string $name): bool
+    {
+        $user = $this->createQueryBuilder('user')
+            ->where('user.name = :name')
+            ->setParameter('name', $name)
+            ->getQuery()
+            ->getOneOrNullResult();
+
+        if (null === $user || !$user instanceof User) {
+            return false;
+        }
+
+        return true;
+    }
+
+    public function isEmailAlreadyInUsed(string $email): bool
+    {
+        $user = $this->createQueryBuilder('user')
+            ->where('user.email = :email')
+            ->setParameter('email', $email)
+            ->getQuery()
+            ->getOneOrNullResult();
+
+        if (null === $user || !$user instanceof User) {
+            return false;
+        }
+
+        return true;
+    }
+
+    public function findOneByEmail(string $email): ?User
+    {
+        $user = $this->createQueryBuilder('user')
+            ->where('user.email = :email')
+            ->setParameter('email', $email)
+            ->getQuery()
+            ->getOneOrNullResult();
```

Comment:
D'ailleurs tout ça est remplacable par  : 
```suggestion
        return $this->findOneBy['email' => $email]);
```

Reply:
J'étais pas aller jusqu’à ce commentaire qui m'aurait donné la réponse héhé